apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx

```diff
@@ -104,48 +102,26 @@ function ClearScheduleButton() {
     );
 }
 
-function CopyScheduleButton() {
-    const [scheduleNames, setScheduleNames] = useState(AppStore.getScheduleNames());
+function CopyScheduleButton(props: { index: number }) {
+    const [open, setOpen] = useState(false);
 
-    useEffect(() => {
-        /**
-         * A shallow copy needs to be made so the array reference is different and the component re-renders.
-         */
-        const handleScheduleNamesChange = () => {
-            setScheduleNames([...AppStore.getScheduleNames()]);
-        };
-
-        AppStore.on('scheduleNamesChange', handleScheduleNamesChange);
+    const handleOpen = useCallback(() => {
+        setOpen(true);
+    }, []);
 
-        return () => {
-            AppStore.off('scheduleNamesChange', handleScheduleNamesChange);
-        };
+    const handleClose = useCallback(() => {
+        setOpen(false);
     }, []);
 
     return (
-        <PopupState variant="popover">
-            {(popupState) => (
-                <>
-                    <Tooltip title="Copy Schedule">
-                        <IconButton {...bindTrigger(popupState)} sx={buttonSx} size="medium">
-                            <ContentCopy />
-                        </IconButton>
-                    </Tooltip>
-                    <Menu {...bindMenu(popupState)}>
-                        {scheduleNames.map((name, index) => (
-                            <MenuItem
-                                key={index}
-                                disabled={AppStore.getCurrentScheduleIndex() === index}
-                                onClick={createCopyHandler(index)}
-                            >
-                                Copy to {name}
-                            </MenuItem>
-                        ))}
-                        <MenuItem onClick={createCopyHandler(scheduleNames.length)}>Copy to All Schedules</MenuItem>
-                    </Menu>
-                </>
-            )}
-        </PopupState>
+        <>
+            <Tooltip title="Copy Schedule">
+                <IconButton sx={buttonSx} onClick={handleOpen} size="small">
+                    <ContentCopy />
+                </IconButton>
+            </Tooltip>
+            <CopyScheduleDialog fullWidth open={open} index={props.index} onClose={handleClose} />
```

Comment:
```suggestion
            <CopyScheduleDialog fullWidth open={open} index={index} onClose={handleClose} />
```
Follow-up change from above w/ props